OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java

```diff
@@ -293,21 +298,36 @@ public void onUserLoginSuccess(ProfileModel profile) {
     public void onUserLoginFailure(Exception ex, String accessToken, String backend) {
         tryToSetUIInteraction(true);
 
-
-        // handle if this is a LoginException
         if (ex != null && ex instanceof LoginException) {
-            LoginErrorMessage error = (((LoginException) ex).getLoginErrorMessage());
-
+            LoginErrorMessage errorMessage = (((LoginException) ex).getLoginErrorMessage());
             showAlertDialog(
-                    error.getMessageLine1(),
-                    (error.getMessageLine2() != null) ?
-                            error.getMessageLine2() : getString(R.string.login_failed));
+                    errorMessage.getMessageLine1(),
+                    (errorMessage.getMessageLine2() != null) ?
+                            errorMessage.getMessageLine2() : getString(R.string.login_failed));
+        } else if (ex != null && ex instanceof HttpStatusException &&
+                ((HttpStatusException) ex).getStatusCode() == HttpStatus.UPGRADE_REQUIRED) {
+            showAppUpdateDialog();
         } else {
-            showAlertDialog(getString(R.string.login_error), getString(R.string.error_unknown));
+            showAlertDialog(getString(R.string.login_error), ErrorUtils.getErrorMessage(ex, LoginActivity.this));
             logger.error(ex);
         }
     }
 
+    public void showAppUpdateDialog() {
```

Comment:
The same function is copied in `RegisterActivity`, we should centralize the logic.